spec/API_specification/signatures/creation_functions.py

```diff
@@ -0,0 +1,370 @@
+from ._types import (List, NestedSequence, Optional, SupportsBufferProtocol, Tuple, Union, array,
+                     device, dtype)
+
+def arange(start: Union[int, float], /, stop: Optional[Union[int, float]] = None, step: Union[int, float] = 1, *, dtype: Optional[dtype] = None, device: Optional[device] = None) -> array:
+    """
+    Returns evenly spaced values within the half-open interval ``[start, stop)`` as a one-dimensional array.
+
+    Parameters
+    ----------
+    start: Union[int, float]
+        if ``stop`` is specified, the start of interval (inclusive); otherwise, the end of the interval (exclusive). If ``stop`` is not specified, the default starting value is ``0``.
+    stop: Optional[Union[int, float]]
+        the end of the interval. Default: ``None``.
+    step: Union[int, float]
+        the distance between two adjacent elements (``out[i+1] - out[i]``). Must not be ``0``; may be negative, this results in an empty array if ``stop >= start``. Default: ``1``.
+    dtype: Optional[dtype]
+        output array data type. If ``dtype`` is ``None``, the output array data type must be inferred from ``start``, ``stop`` and ``step``. If those are all integers, the output array dtype must be the default integer dtype; if one or more have type ``float``, then the output array dtype must be the default floating-point data type. Default: ``None``.
+    device: Optional[device]
+        device on which to place the created array. Default: ``None``.
+
+    Returns
+    -------
+    out: array
+        a one-dimensional array containing evenly spaced values. The length of the output array must be ``ceil((stop-start)/step)`` if ``stop - start`` and ``step`` have the same sign, and length ``0`` otherwise.
+
+    Notes
+    -----
+    - This function cannot guarantee that the interval does not include the ``stop`` value in those cases where ``step`` is not an integer and floating-point rounding errors affect the length of the output array.
```

Comment:
Can we keep this formatted with the `note` directive? I don't know if it's possible to keep it in its original location next to the `stop` parameter, but if that's possible it would be preferable.

Reply:
I tried, but the issue is that I cannot mix rst syntax given that the docstring is in numpy format. So, it just doesn't render as expected 😕